backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/NokiaInnovativeProjectApplication.java

```diff
@@ -30,4 +34,20 @@ public CorsFilter corsFilter() {
 		source.registerCorsConfiguration("/**", config);
 		return new CorsFilter(source);
 	}
+
+    @Profile("production")
+	@Bean
```

Comment:
What is the purpose of this configuration?
You can easily configure it in yml files.

Reply:
After configuration was moved to yml, app crashed. It seems like Heroku does not set JBDC_DATABASE_URL var.
![image](https://user-images.githubusercontent.com/30317283/49219581-03f7fc00-f3d4-11e8-90c5-7839adf409f7.png)

Reply:
What could cause this kind of problem?

Reply:
Unsatisfied dependency (..). It means that Spring cannot inject some dependencies.
Did you specify all configuration properties?
Can you provide me your code on some branch?

Reply:
All code is on this branch.

Reply:
A lot of people have a simular problem. But they are just spelling database URL wrong and keeps getting something like this : 
`Driver org.postgresql.Driver claims to not accept jdbcUrl, jdbc:postgres://iqmfgfnq:BmIRuw3i1_mWoOBbqI_jPIsbKSxy-aHY@horton.elephantsql.com:5432/iqmfgfnq`
But in our exception, we are getting ${JDBC_DATABASE_URL} instead of actual URL. That's why I think that is a Heroku problem. It messes up with setting this URL.

Reply:
You are getting ${JDBC_DATABASE_URL} because environment variable is not set. 
Remember that you are using Docker image. You have separated environment for your application. You have to set somehow JDBC_DATABASE_URL. 
Instead using env variables you can encrypt your sensitive data: 
https://www.baeldung.com/spring-boot-jasypt

Reply:
So I can return that Bean with basicDataSource in NokiaInnovativeProjectApplication?